src/extension/csw/core/src/main/java/org/geoserver/csw/store/internal/CatalogStoreMapping.java

```diff
@@ -229,6 +231,13 @@ public static CatalogStoreMapping parse(Map<String, String> mappingSource) {
             }
         }
 
+        int index = mappingName.indexOf('-');
+        if (index >= 0) {
+            mapping.setMappingName(mappingName.substring(index + 1));
+        } else {
+            mapping.setMappingName("");
+        }
```

Comment:
Could you clarify what is happening here? Also, I don't see any documentation update on how to use the multiple mappings facility.

Reply:
The mappings are given a name based on the filename: Record-thefirst.properties gets the name 'thefirst'. 

Documentation on the way